tools/dm/practice.md

```diff
@@ -0,0 +1,176 @@
+---
+title: 使用 DM 同步数据
+category: tools
+---
+
+# 使用 DM 同步数据
+
+本文介绍如何使用 DM (Data Migration) 同步数据。
+
+## 第 1 步：部署 DM 集群
+
+目前推荐使用 DM-Ansible 部署 DM 集群，具体部署方法参照 [使用 DM-Ansible 部署 DM](/tools/dm/deployment.md)。
+
+> **注意**：
+> 
+> - 在 DM 所有的配置文件中，数据库的密码要使用 dmctl 加密后的密文。如果数据库密码为空，则不需要加密。关于如何使用 dmctl 加密明文密码，参考[使用 dmctl 加密上游 MySQL 用户密码](/tools/dm/deployment.md#使用-dmctl-加密上游-mysql-用户密码)。
+> - 上下游数据库用户必须拥有相应的读写权限。
+
+## 第 2 步：检查集群信息
+
+使用 DM-Ansible 部署 DM 集群后，相关配置信息如下：
+
+- DM 集群相关组件配置信息
+
+    | 组件 | 主机 | 端口 |
+    |:------|:---- |:---- |
+    | dm_worker1 | 172.16.10.72 | 8262 |
+    | dm_worker2 | 172.16.10.73 | 8262 |
+    | dm_master | 172.16.10.71 | 8261 |
+
+- 上下游数据库实例相关信息
+
+    | 数据库实例 | 主机 | 端口 | 用户名 | 加密密码 |
+    |:-------- |:--- | :--- | :--- | :--- |
+    | 上游 MySQL-1 | 172.16.10.81 | 3306 | root | VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU= |
+    | 上游 MySQL-2 | 172.16.10.82 | 3306 | root | VjX8cEeTX+qcvZ3bPaO4h0C80pe/1aU= |
+    | 下游 TiDB | 172.16.10.83 | 4000 | root | |
+
+- dm-master 进程配置文件 `{ansible deploy}/conf/dm-master.toml` 中的配置
+
+    ```toml
+    # Master 配置
+
+    [[deploy]]
+    source-id = "mysql-replica-01"
+    dm-worker = "172.16.10.72:8262"
+
+    [[deploy]]
+    source-id = "mysql-replica-02"
+    dm-worker = "172.16.10.73:8262"
+    ```
+
+    > **注意**：`{ansible deploy}/conf/dm-master.toml` 中的 `{ansible deploy}` 表示 DM-Ansible 部署的目录。该目录使用 `deploy_dir` 参数配置。
+
+## 第 3 步：配置任务
+
+假设需要将 MySQL-1 和 MySQL-2 实例的 `test_db` 库的 `test_table` 表以**全量+增量**的模式同步到下游 TiDB 的 `test_db` 库的 `test_table` 表。
+
+复制并编辑 `{ansible deploy}/conf/task.yaml.example`，生成如下任务配置文件 `task.yaml`：
+
+```yaml
+# 任务名，多个同时运行的任务不能重名。
+name: "test"
+# 全量+增量 (all) 同步模式。
+task-mode: "all"
+# 下游 TiDB 配置信息。
+target-database:
+  host: "172.16.10.83"
+  port: 4000
+  user: "root"
+  password: ""
+
+# 当前数据同步任务需要的全部上游 MySQL 实例配置。
+mysql-instances:
+-
+  # 上游实例或者复制组 ID，参考 `inventory.ini` 的 `source_id` 或者 `dm-master.toml` 的 `source-id 配置`。
+  source-id: "mysql-replica-01"
+  # 需要同步的库名或表名的黑白名单的配置项名称，用于引用全局的黑白名单配置，全局配置见下面的 `black-white-list` 的配置。
+  black-white-list: "global"
+  # Mydumper 的配置项名称，用于引用全局的 mydumper 配置。
```

Comment:
```suggestion
  # mydumper 的配置项名称，用于引用全局的 mydumper 配置。
```
should `mydumper` keeps the official capitalization. I notice in English doc it is used as `mydumper` at the beginning of sentence.

Reply:
@amyangfei 
I find that when "mydumper" is at the beginning of the sentence, sometimes it is "mydumper" while sometimes it is "Mydumper". OK. Let's use "mydumper" consistently.